src/main/java/com/github/legoatoom/connectiblechains/client/ChainPacketHandler.java

```diff
@@ -0,0 +1,194 @@
+package com.github.legoatoom.connectiblechains.client;
+
+import com.github.legoatoom.connectiblechains.chain.ChainLink;
+import com.github.legoatoom.connectiblechains.chain.ChainType;
+import com.github.legoatoom.connectiblechains.chain.ChainTypesRegistry;
+import com.github.legoatoom.connectiblechains.chain.IncompleteChainLink;
+import com.github.legoatoom.connectiblechains.enitity.ChainCollisionEntity;
+import com.github.legoatoom.connectiblechains.enitity.ChainKnotEntity;
+import com.github.legoatoom.connectiblechains.util.NetworkingPackets;
+import com.github.legoatoom.connectiblechains.util.PacketBufUtil;
+import it.unimi.dsi.fastutil.objects.ObjectArrayList;
+import it.unimi.dsi.fastutil.objects.ObjectList;
+import net.fabricmc.fabric.api.client.networking.v1.ClientPlayNetworking;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.EntityType;
+import net.minecraft.util.math.Vec3d;
+import net.minecraft.util.registry.Registry;
+
+import java.util.UUID;
+
+public class ChainPacketHandler {
```

Comment:
Should be in client environment, no?

Aditionally, this class can be simplified by using creating a function that both the single and the multi call.
Cleaner code and easier to tweak.

Also I see lots of throws, better use logging errors instead.

Reply:
It's already in the client package so adding `@Environment(EnvType.CLIENT)` shouldn't be necessary, as the other classes in this package also don't have that annotation.

Reply:
I've added a `createLinks` and `createEntity` function